### PR TITLE
Add call to create PR in AzDo servicing.

### PR DESCRIPTION
**Pull request by crummel**
Fixes https://github.com/dotnet/source-build/issues/3069.

Release pipieline dry run is at https://dnceng.visualstudio.com/internal/_build/results?buildId=2138107&view=results.

### CODE REVIEW

**Comment by MichaelSimons** [comment]
@crummel - It appears that the dry-run you linked to failed prior to invoking the affected code.  Could you please link to a PR created by these changes?

**Comment by crummel** [comment]
I did end up having an auth problem and having to do some debugging - I've pushed those changes.  The internal CI ran https://dnceng.visualstudio.com/internal/_build/results?buildId=2139683&view=logs&j=3a04dffd-22cf-5707-fa41-9bd4708cc1db&t=102abc8f-88d1-56a3-f0d3-e5f498d0f4f9 and produced the PR https://dev.azure.com/dnceng/internal/_git/security-partners-dotnet/pullrequest/30092.

File: eng/push-tarball.sh [hunk 1]
```diff
@@ -109,11 +109,26 @@ pushd "$vmr_path"
   git add -f .
   git diff --staged --quiet || git commit -m "Update to .NET $sdk_version"
 
+  pr_url=$(echo $upstream_url | sed "s,_git.*,_apis/git/repositories/security-partners-dotnet/pullrequests?api-version=7.0,g")
+  data="{
+    \"sourceRefName\": \"refs/heads/$new_branch_name\",
+    \"targetRefName\": \"refs/heads/$target_branch\",
+    \"title\": \"Source-build update to $sdk_version\",
```
Comment (MichaelSimons) [inline eng/push-tarball.sh:116]: Previous PRs were using this title.  Not sure there is value one way or the other but for consistency I don't see a reason to change, do you?

```suggestion
    \"title\": \"Update to $sdk_version\",
```

**Review by MichaelSimons — COMMENTED** [review]
One minor suggestion/question.  The rest looks good.

**Review by MichaelSimons — APPROVED** [review]
Friendly reminder to cleanup the internal PR as part of merging.

**Pull request merged**